internal/talk/talk.go

```diff
@@ -0,0 +1,129 @@
+package talk
+
+// Talk compiles all the functionality related with the Talk product.
+// It is used by Corald to massage the data before sending to all other services it uses.
+
+import (
+	"errors"
+
+	"github.com/coralproject/shelf/internal/sponge/item"
+)
+
+var (
+	// ErrActionNotFound is action not found error.
+	ErrActionNotFound = errors.New("Action not found")
+
+	//ErrActionNotAllowed is an error when the action is not allowed.
+	ErrActionNotAllowed = errors.New("Action not allowed")
+
+	//ErrTypeNotExpected comes when the type asserted was not expected
+	ErrTypeNotExpected = errors.New("Type not expected")
+)
+
+//==============================================================================
+
+// AddAction an action (flag, like, etc) on an Item.
+func AddAction(spongedURL string, user item.Item, action string, targetID string) (item.Item, error) {
+
+	// The action must be in an allowed actions list.
+	// To Do: Is this user allowed to do this action on this target?
+	if err := userAllowed(user, action); err != nil {
+		return item.Item{}, err
+	}
+
+	// Users may only add on action of a certain type to one item.
```

Comment:
Removed. This PR is missing the authorization part of users on actions. That needs a little more discussion.